cloudstack.go

```diff
@@ -944,8 +975,12 @@ func (d *Driver) associatePublicIP() error {
 	log.Infof("Associating public ip address...")
 	p := cs.Address.NewAssociateIpAddressParams()
 	p.SetZoneid(d.ZoneID)
-	if d.NetworkID != "" {
-		p.SetNetworkid(d.NetworkID)
+	if len(d.NetworkID) > 0 {
+		if d.PublicInterfaceIndex < len(d.NetworkID) {
```

Comment:
You could invert the condition and remove the `else`.

Reply:
Done!